src/games/squarer.js

```diff
@@ -0,0 +1,220 @@
+const { Colors, GameNode } =  require('squishjs');
+const Game = require('./Game');
+const { randomColor, BLACK, GRAY, GOLD, GREEN } = Colors;
+
+class Squarer extends Game {
+	static metadata() {
+        return {
+            res: {
+                width: 1920,
+                height: 1080
+            },
+            author: 'Yazeed Loonat'
+        };
+    }
+
+	constructor() {
+        super();
+        this.defaultSize = { x: 5, y: 5 };
+		this.base = GameNode(BLACK, null, {x: 0, y: 0}, {x: 100, y: 100});
+        this.startLine = GameNode(GRAY, null, {x: 0, y: 95}, {x: 100, y: 5});
+        this.finishLine = GameNode(GRAY, null, {x: 0, y: 0}, {x: 100, y: 5});
+        this.base.addChild(this.startLine);
+        this.base.addChild(this.finishLine);
+
+        const destinationColor = randomColor([BLACK, GRAY, GOLD, GREEN]);
+        this.destination = GameNode(destinationColor, null, {x: 40, y: 0}, this.defaultSize);
+        this.base.addChild(this.destination);
+
+        this.keyCoolDowns = {};
+        this.level = 1;
+        this.usedColors = [BLACK, GRAY, GOLD, GREEN, destinationColor];
+        this.playerArray = [];
+        this.movementQueue = [];
+        this.npc = [];
+        this.createNpc();
+    }
+
+    getRoot() {
+        return this.base;
+    }
+
+    createNpc() {
+        this.movementQueue = [];
+        this.npc.forEach(npc => {
+            clearInterval(npc.interval);
+            this.base.removeChild(npc.id);
+        });
+        this.npc = [];
+        if (this.level > 9) {
+            console.log("good job");
+            this.base.removeChild(this.destination.id)
+            return;
+        }
+        for( let i = 5; i < 50; i+=5) {
+            const tickRate = Math.ceil(250 + Math.random() * 100);
+            let color = i % 2 ? GOLD : GREEN;
+            for (let l = 0; l < this.level && l <= 9; l++) {
+                const temp = GameNode(color, null, {x: (10 * l) + 5, y: i}, this.defaultSize);
+                temp.interval = setInterval(() => {
+                    this.movementQueue.push({ npc: true, node: temp, direction: 'left' });
+                }, tickRate);
+                this.npc.push(temp);
+                this.base.addChild(temp);
+            }
+        }
+
+        for( let i = 55; i > 50 && i < 95; i+=5) {
```

Comment:
Shouldn't `i` _always_ be above 50 in this loop?

Reply:
nice catch, updated, thank you!